doc/manual/metaprogramming.rst

```diff
@@ -1028,12 +1037,40 @@ copy them*, for the following reasons:
 * the ``foo`` function has side-effects, and it is undefined exactly when,
```

Comment:
As `@mauro3` suggested in https://github.com/JuliaLang/julia/pull/18679/files#r80905058, perhaps also mention the issues immediately alongside the corresponding examples?